### PR TITLE
WT-8423 Add flush timestamp to tiered and object metadata.

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

File: src/conn/conn_dhandle.c [hunk 1]
```diff
@@ -100,7 +100,9 @@ __conn_dhandle_config_set(WT_SESSION_IMPL *session)
         cfg[0] = tmp;
         cfg[1] = NULL;
         if (dhandle->type == WT_DHANDLE_TYPE_TIERED)
-            strip = "checkpoint=,checkpoint_backup_info=,checkpoint_lsn=,last=,tiers=()";
+            strip =
+              "checkpoint=,checkpoint_backup_info=,checkpoint_lsn=,flush_time=,flush_timestamp,"
```
Comment (ddanderson) [inline src/conn/conn_dhandle.c:104]: Shouldn't this be `flush_timestamp=`?

File: test/suite/test_tiered11.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        end_ts = self.timestamp_str(end-1)
+        self.conn.set_timestamp('oldest_timestamp=' + end_ts + ',stable_timestamp=' + end_ts)
+        c.close()
+        return end_ts
+
+    # Test calling the flush_tier API.
+    def test_tiered11(self):
+        # Create three tables. One using the system tiered storage, one
+        # specifying its own bucket and object size and one using no
+        # tiered storage. Use stats to verify correct setup.
```
Comment (ddanderson) [inline test/suite/test_tiered11.py:98]: Update comment?

**Review by ddanderson — COMMENTED** [review]
I think the missing `=` in the config might be significant.

File: src/conn/conn_dhandle.c (hunk 1 above)
Comment (sueloverso) [reply src/conn/conn_dhandle.c:104]: Good catch. Fixed.

File: test/suite/test_tiered11.py (hunk 2 above)
Comment (sueloverso) [reply test/suite/test_tiered11.py:98]: Fixed.

**Review by ddanderson — APPROVED** [review]
LGTM

**Review by keitharnoldsmith — COMMENTED** [review]
It looks like we could wind up storing different flush timestamps in the `tiered:` metadata and the `object:` metadata.  In both cases the timestamp comes from `conn.txn_global.last_ckpt_timestamp`.  But it seems that value could change between calling `flush_tier_once()` and `tier_flush_meta()`.

Is this intentional?

**Comment by sueloverso** [comment]
I will open a new ticket - as this is a valid point. The tiered metadata gets updated when the objects are copied and needs to have the same. Sorry to have jumped the gun.

**Pull request merged**